app/src/components/PopularVendor.tsx

```diff
@@ -1,46 +1,44 @@
-import { useEffect, useState } from "react";
-import { usePopularVendorQuery } from "../api";
-import {
-  BarChart,
-  Bar,
-  XAxis,
-  YAxis,
-  CartesianGrid,
-  Tooltip,
-  Legend,
-} from "recharts";
+//with query
+import React, { useEffect, useState } from "react";
+import { usePopularVendorQuery, PopularVendor } from "../api";
+import { BarChart, Bar, XAxis, YAxis, Tooltip, Legend } from "recharts";
 
 export default function PopularVendor() {
-  const { data: area } = usePopularVendorQuery();
+  const { data: popularVendor } = usePopularVendorQuery();
   const [vendorByArea, setVendorByArea] = useState([] as any);
-
   useEffect(() => {
-    if (area) {
-      let obj = {
-        ID: "30f15697-8c12-41cf-83d9-a9ccfe07c43e",
-        name: "Areas",
-        /* "Fish Taco": area.BusinessName,
-                "Bellevue": area.AreaName,
-                "TotalRating": area.One,*/
-      };
+    if (popularVendor) {
+      let temp = [] as PopularVendor[];
+      for (const Vendor of popularVendor) {
+        let obj = {
+          TotalRatings: Vendor.TotalRatings,
+          BusinessName: Vendor.BusinessName,
+          Location: Vendor.Location,
+        };
+
+        temp.push(obj);
+      }
 
-      setVendorByArea([obj]);
+      setVendorByArea(temp);
     }
-  }, [area]);
+  }, [popularVendor]);
+
+  console.log(vendorByArea);
 
   return (
-    <BarChart width={800} height={500} data={vendorByArea}>
-      /* <CartesianGrid strokeDasharray="3 3" />
-      */
-      <XAxis dataKey="Name" />
-      <YAxis />
+    <BarChart
+      width={600}
+      height={400}
+      data={popularVendor}
+      layout={"vertical"}
+      margin={{ top: 20, right: 30, left: 30, bottom: 5 }}
+    >
+      <XAxis type={"number"} />
+      <YAxis type="category" dataKey="BusinessName" />
       <Tooltip />
       <Legend />
-      <Bar dataKey="Name" fill="#FF6347" width={1} />
-      <Bar dataKey="Name" fill="#0000FF" />
-      <Bar dataKey="Name" fill="#FF00FF" />
-      <Bar dataKey="Name" fill="#4B0082" />
-      <Bar dataKey="Name" fill="#FFA500" />
+      <Bar dataKey={"TotalRatings"} fill="#FF6347" />
+      <Bar dataKey="Location" fill="#FFA500" />
```

Comment:
The Location column should not be a second bar. There should only be a single TotalRating bar.

Reply:
I did try that but when I remove the second bar the graph won't render

Reply:
![Screenshot from 2022-05-19 09-51-46](https://user-images.githubusercontent.com/36318069/169355275-4a80898e-34ab-41f6-a6c9-71f7a424c76e.png)

By removing the Location Bar, it results in a single TotalRating bar.

Reply:
The problem is when you take it off and you hover over you wont see the location and this is one of the creteria.

Reply:
![Picture1](https://user-images.githubusercontent.com/36318069/169360085-27679f3f-a000-4c65-a89b-cf5982f63d5c.png)

There is only a single category in the bar graph design, with no mention of hovering over the graph. Since only a single location should be shown, there is no use in showing the location for each data point.

Reply:
On top is the filter for Location, and as we are not having the filters we will be able to see the Location. Unless we don't want to see the location at all. We did make some adjustments to the graphs when I presented the queries to the team.